diff -ruN a/src/Container.php b/src/Container.php
--- a/src/Container.php
+++ b/src/Container.php
@@ -47,9 +47,9 @@
      * @param InflectorAggregateInterface|null       $inflectors
      */
     public function __construct(
-        DefinitionAggregateInterface      $definitions = null,
-        ServiceProviderAggregateInterface $providers = null,
-        InflectorAggregateInterface       $inflectors = null
+        ?DefinitionAggregateInterface      $definitions = null,
+        ?ServiceProviderAggregateInterface $providers = null,
+        ?InflectorAggregateInterface       $inflectors = null
     ) {
         $this->definitions = $definitions ?? new DefinitionAggregate;
         $this->providers   = $providers   ?? new ServiceProviderAggregate;
@@ -77,7 +77,7 @@
      *
      * @return DefinitionInterface
      */
-    public function add(string $id, $concrete = null, bool $shared = null) : DefinitionInterface
+    public function add(string $id, $concrete = null, ?bool $shared = null) : DefinitionInterface
     {
         $concrete = $concrete ?? $id;
         $shared = $shared ?? $this->defaultToShared;
@@ -222,7 +222,7 @@
      *
      * @return InflectorInterface
      */
-    public function inflector(string $type, callable $callback = null) : InflectorInterface
+    public function inflector(string $type, ?callable $callback = null) : InflectorInterface
     {
         return $this->inflectors->add($type, $callback);
     }
diff -ruN a/src/Inflector/Inflector.php b/src/Inflector/Inflector.php
--- a/src/Inflector/Inflector.php
+++ b/src/Inflector/Inflector.php
@@ -37,7 +37,7 @@
      * @param string        $type
      * @param callable|null $callback
      */
-    public function __construct(string $type, callable $callback = null)
+    public function __construct(string $type, ?callable $callback = null)
     {
         $this->type     = $type;
         $this->callback = $callback;
diff -ruN a/src/Inflector/InflectorAggregate.php b/src/Inflector/InflectorAggregate.php
--- a/src/Inflector/InflectorAggregate.php
+++ b/src/Inflector/InflectorAggregate.php
@@ -17,7 +17,7 @@
     /**
      * {@inheritdoc}
      */
-    public function add(string $type, callable $callback = null) : Inflector
+    public function add(string $type, ?callable $callback = null) : Inflector
     {
         $inflector          = new Inflector($type, $callback);
         $this->inflectors[] = $inflector;
diff -ruN a/src/Inflector/InflectorAggregateInterface.php b/src/Inflector/InflectorAggregateInterface.php
--- a/src/Inflector/InflectorAggregateInterface.php
+++ b/src/Inflector/InflectorAggregateInterface.php
@@ -15,7 +15,7 @@
      *
      * @return Inflector
      */
-    public function add(string $type, callable $callback = null) : Inflector;
+    public function add(string $type, ?callable $callback = null) : Inflector;
 
     /**
      * Applies all inflectors to an object.
